Bio/SeqIO/QualityIO.py

```diff
@@ -796,6 +796,59 @@ def _get_solexa_quality_str(record):
                    for qp in qualities)
 
 
+def FastqStrictIterator(handle):
+    """Iterate over strictly four line Fastq records as string tuples.
+
+    In contrast to the FastqGeneralIterator, this iterator does allow for the
+    splitting of records over multiple lines.  Line wrapping is out of vogue and
+    hence this iterator provides a faster but stricter alternative to the
+    FastqGeneralIterator.
+
+    Iterate over Fastq records as string tuples (not as SeqRecord objects).
+
+    This code does not try to interpret the quality string numerically.  It
+    just returns tuples of the title, sequence and quality as strings.  For
+    the sequence and quality, any whitespace (such as new lines) is removed.
+
+    Our SeqRecord based FASTQ iterators call this function internally, and then
+    turn the strings into a SeqRecord objects, mapping the quality string into
+    a list of numerical scores.  If you want to do a custom quality mapping,
+    then you might consider calling this function directly.
+
+    Because we now know that the records have four lines each, we can simply
+    iterate over groups of four lines without having to worry about identifying
+    what each line is.
+    """
+    for line in handle:
+        if line == '\n' or line == '':
+                continue
+        try:
+            line_1 = line.strip()
+            line_2 = next(handle).strip()
+            line_3 = next(handle).strip()
+            line_4 = next(handle).strip()
+
+            # check if file is in binary format
+            if isinstance(line_1[0], int):
```

Comment:
This check would happen for every single record - it seems redundant and a potential speed overhead.

Reply:
I agree; I've tried something else which does avoid this problem but makes the code look slightly bulky(?) IMO. I treat the first record separately from the rest of them, so there is some repetition. Let me know what you think.